### PR TITLE
Issue/13329 backup details unit tests

### PR DESCRIPTION
**Pull request by zwarm**
Parent #13329 

This PR adds unit tests for `BackupDownloadDetailsViewModel` and `BackupAvailableItemsProvider`

To test:
Run tests

PR submission checklist:

- [X] I have considered adding unit tests where possible.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/BackupDownloadDetailsViewModelTest.kt [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        )
+    }
+
+    @Test
+    fun `when available items are fetched, the content view is shown`() = test {
+        val uiStates = initObservers().uiStates
+
+        viewModel.start()
+
+        assertThat(uiStates[0]).isInstanceOf(UiState.Content::class.java)
```
Comment (ParaskP7) [inline WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/BackupDownloadDetailsViewModelTest.kt]: Minor (🔍 ): The explicit `UiState` can be omitted, or the `UiState.Content` import can be removed and the `UiState` can be made explicit in places required.

File: WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/BackupDownloadDetailsViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+    @Test
+    fun `item is unchecked, when it is clicked on`() = test {
+        val uiStates = initObservers().uiStates
+
+        viewModel.start()
+
+        ((uiStates.last() as Content).items[1]).onClick.invoke()
+
+        assertThat(((uiStates.last() as Content).items[1]).checked).isFalse()
```
Comment (ParaskP7) [inline WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/BackupDownloadDetailsViewModelTest.kt:47]: Minor (🔍 ): Consider a property access syntax as suggest here by the IDE.

PS: The same apply for the `.isTrue()` in the below test.

File: WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/BackupDownloadDetailsViewModelTest.kt [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    @Test
+    fun `item is checked, when it is clicked on twice `() = test {
```
Comment (ParaskP7) [inline WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/BackupDownloadDetailsViewModelTest.kt]: Idea (💡 ): These kind of tests scenarios can benefit from a `given/when/then` pattern. Utilising this pattern, I would have named this test case as: `given item is unchecked, when item is clicked, then item gets checked`. Also I would have asserted the `unchecked` state in the setup, like with the `.isFalse()` above and then, trigger the next `.invoke()` action separately and check the `checked` state afterwards.

**Review by ParaskP7 — COMMENTED** [review]
Thank you again @zwarm ! 🌟 

Only minors from my sides.

File: WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/BackupDownloadDetailsViewModelTest.kt (hunk 2 above)
Comment (zwarm) [reply WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/BackupDownloadDetailsViewModelTest.kt:47]: @ParaskP7 - interesting, but my IDE is not suggesting property access syntax. 🤷

Comment (ParaskP7) [reply WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/BackupDownloadDetailsViewModelTest.kt:47]: @zwarm - Can you check your Kotlin version? We can sit down together to find out why is that.

**Comment by ashiagr** [comment]
Thanks @zwarm! Tests look fine for the current functionality. We can merge this PR once comments mentioned by @ParaskP7 are resolved.

**Review by ashiagr — APPROVED** [review]

**Pull request merged**